core/src/test/java/fr/sncf/osrd/envelope_sim/AllowanceRangesTests.java

```diff
@@ -237,4 +241,58 @@ public void testVeryShortRange() {
         assertEquals(marginTime2, targetTime2, 2 * TIME_STEP);
         assertEquals(marginTime3, targetTime3, 2 * TIME_STEP);
     }
+
+    /** Test with an allowance range that starts very slightly after the path start, and ends around the end
+     * of the acceleration part. This doesn't necessarily have to result in a valid envelope as we're very close
+     * to asking for an impossible allowance, but we check that it doesn't crash early */
+    @Test
+    public void testAllowanceRangeEdgeCase() {
+        var testRollingStock = TestTrains.REALISTIC_FAST_TRAIN;
+        var length = 100_000;
+        var testPath = new FlatPath(length, 0);
+        var testContext = new EnvelopeSimContext(testRollingStock, testPath, TIME_STEP);
+        var stops = new double[] { testPath.getLength() };
+        var maxEffortEnvelope = makeSimpleMaxEffortEnvelope(testContext, 10, stops);
+        var value1 = new AllowanceValue.Percentage(10);
+        var rangesTransitions = new double[] { 0.1, maxEffortEnvelope.get(0).getEndPos() };
+        var ranges = List.of(
+                new AllowanceRange(rangesTransitions[0], rangesTransitions[1], value1)
+        );
+        var allowance = new LinearAllowance(
+                testContext, rangesTransitions[0], rangesTransitions[1], 0, ranges
+        );
+        applyAllowanceIgnoringUserError(allowance, maxEffortEnvelope);
+    }
+
+    /** Test with an allowance range that starts very slightly after the path start, and ends around the end
+     * of the acceleration part, with a large time step.
+     * This doesn't necessarily have to result in a valid envelope as we're very close
+     * to asking for an impossible allowance, but we check that it doesn't crash early */
+    @Test
+    public void testAllowanceRangeEdgeCaseLargeTimeStep() {
+        var testRollingStock = TestTrains.REALISTIC_FAST_TRAIN;
+        var length = 100_000;
+        var testPath = new FlatPath(length, 0);
+        var testContext = new EnvelopeSimContext(testRollingStock, testPath, TIME_STEP * 2);
+        var stops = new double[] { testPath.getLength() };
+        var maxEffortEnvelope = makeSimpleMaxEffortEnvelope(testContext, 10, stops);
+        var value1 = new AllowanceValue.Percentage(10);
+        var rangesTransitions = new double[] { 0.1, maxEffortEnvelope.get(0).getEndPos() };
+        var ranges = List.of(
+                new AllowanceRange(rangesTransitions[0], rangesTransitions[1], value1)
+        );
+        var allowance = new LinearAllowance(
+                testContext, rangesTransitions[0], rangesTransitions[1], 0, ranges
+        );
+        applyAllowanceIgnoringUserError(allowance, maxEffortEnvelope);
+    }
+
+    /** Applies the allowance to the envelope. Any user error (impossible margin) is ignored */
```

Comment:
I don't get why user errors should be ignored. Shouldn't we know what the expected result is for this specific case?

Reply:
Now that the bug is fixed we can see what the result is, but it's not what the test is about. I don't want this test to fail if someday we change the way we compute allowances in one way or another, the result is irrelevant